### PR TITLE
Bump diesel to 2.0

### LINKED ISSUE #2862: Upgrade diesel to 2.0
### Description and goal

The new search feature needs some feature that aren't available until 2.0.
See the [migration guide](https://diesel.rs/guides/migration_guide.html)

### Acceptance criteria

- [ ] rocket was updated to 2.0
- [ ] tests pass

### PR DESCRIPTION
**Pull request by leovalais**
Rocket has not released Diesel 2.0 support on crates.io yet.
I also bumped rocket_cors' rev (no breaking change).

Closes #2862

### CODE REVIEW

File: editoast/Cargo.toml [hunk 1]
```diff
@@ -14,15 +14,17 @@ clap = { version = "~4.0.26", features = ["derive", "env"] }
...
-rocket = { version = "0.5.0-rc", features = ["json"] }
-rocket_sync_db_pools = { version = "0.1.0-rc", features = [
+rocket = { git = "https://github.com/SergioBenitez/Rocket.git", rev = "162fafa0", features = [
+    "json",
+] }
+rocket_sync_db_pools = { git = "https://github.com/SergioBenitez/Rocket.git", rev = "162fafa0", features = [
     "diesel_postgres_pool",
 ] }
-rocket_cors = { git = "https://github.com/lawliet89/rocket_cors.git", rev = "54fae070" }
+rocket_cors = { git = "https://github.com/lawliet89/rocket_cors.git", rev = "c17e8145" }
```
Comment (flomonster) [inline editoast/Cargo.toml]: The CI doesn't pass on this commit. https://github.com/lawliet89/rocket_cors/actions/runs/2990544894.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for this PR. I think we should really consider stop using rocket.

File: editoast/Cargo.toml (hunk 1 above)
Comment (Khoyo) [reply editoast/Cargo.toml]: Only on nightly though

Comment (leovalais) [reply editoast/Cargo.toml]: I reverted it anyway since that commit doesn't really adds anything but I agree that the state of rocket isn't ideal...

**Review by flomonster — APPROVED** [review]

**Pull request merged**